From 55713ef94a6ad814eff2790c6ffb793f8d30eb71 Mon Sep 17 00:00:00 2001
From: Richard Gong <richard.gong@intel.com>
Date: Thu, 1 Apr 2021 13:09:06 -0500
Subject: [PATCH 117/120] HSD #16013024105: fpga: stratix10-soc: stop kernel
 thread

commit a8566a09b374956a1e655ef96658bd377a67437d from
https://github.com/altera-opensource/linux-socfpga.git

The kernel thread is still running after stratix10-soc gets the running
firmware version. Stop kernel thread after FW version service is completed
to save the kernel resources.

Signed-off-by: Richard Gong <richard.gong@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/fpga/stratix10-soc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/fpga/stratix10-soc.c b/drivers/fpga/stratix10-soc.c
index 9ab7afd28a64..6ba59841fbf7 100644
--- a/drivers/fpga/stratix10-soc.c
+++ b/drivers/fpga/stratix10-soc.c
@@ -486,10 +486,12 @@ static int s10_probe(struct platform_device *pdev)
 			       NULL, 0, s10_fw_version_callback);
 	if (ret) {
 		dev_err(dev, "couldn't get firmware version\n");
+		stratix10_svc_done(priv->chan);
 		fpga_mgr_free(mgr);
 		goto probe_err;
 	}
 
+	stratix10_svc_done(priv->chan);
 	platform_set_drvdata(pdev, mgr);
 	return ret;
 
-- 
2.31.1

